src/main/java/sdmExam/Game.java

```diff
@@ -0,0 +1,24 @@
+package sdmExam;
+
+public class Game {
+    private final Board board = new Board();
+    private Stone lastPlay = Stone.WHITE;
+
+    public void play(Stone player, Position position) throws Exception {
+
+        if (isInvalidFirstPlayer(player)) {
+            throw new Exception("Black player should play first.");
+        }
+
+        if (board.isOccupied(position)) {
+            throw new Exception("Position is already occupied.");
+        }
+
+        board.addStoneAt(player, position);
+        lastPlay = player;
```

Comment:
One question here, let's say black played first. Later black tried to play again. Obviously, exception "Black should play first." will be thrown, but is that exception correct in every case? (like it is invalid the example that I gave, isn't it?)

Reply:
With "Later black tried to play again" do you mean if black player try to play twice? This case is manage by setting _lastPlay_ with the last player, and the first conditional. Anyway, on _GameShould_ class you can find _notAllowPlayerToPlayTwiceInARow_ test.

Reply:
> With "Later black tried to play again" do you mean if black player try to play twice? This case is manage by setting _lastPlay_ with the last player, and the first conditional. Anyway, on _GameShould_ class you can find _notAllowPlayerToPlayTwiceInARow_ test.

Agree with Pietro

Reply:
I understand how does it work but my concern is about the exception that we throw particularly sentence. In the case that I gave just an example, what will be the message that user see? It will be "Black player should play first." and it doesn't make sense for that case, when we violate that conditional we'll see always "Black player should play first.", or I am confusing? Anyway I can explain better in the meeting